src/App.js

```diff
@@ -1,23 +1,46 @@
-import logo from './logo.svg';
+import { Row } from 'antd';
+import { useState } from 'react';
 import './App.css';
+import { AddFoodForm } from './components/AddFoodForm';
+import { FoodBox } from './components/FoodBox';
+import { SearchBar } from './components/SearchBar';
+import { foods } from './utils';
 
 function App() {
+  const [foodsList, setFoods] = useState(foods);
+  const [searchString, setSearchString] = useState('');
+
+  const handleSearchChange = (e) => {
+    setSearchString(e.target.value);
+  };
+
+  const addFoodItem = (newFoodItem) => {
+    setFoods((old) => {
+      return [...old, newFoodItem];
+    });
+  };
+
+  // const handleDelete = (e) => {
+  //   setFoods((old) => {
+  //     return old.filter((food) => food.key !== e.currentTarget.id);
+  //   });
+  // };
+
+  const handleDelete = (e) => {
+    let newArr = [...foodsList].filter(
+      (food) => food.key !== e.currentTarget.id
```

Comment:
just know, there is a slight difference between currentTarget and target. they are not always, exactly the same: https://stackoverflow.com/questions/10086427/what-is-the-exact-difference-between-currenttarget-property-and-target-property